xarray/backends/netCDF4_.py

```diff
@@ -485,21 +496,66 @@ def encode_variable(self, variable):
         return variable
 
     def prepare_variable(
-        self, name, variable, check_encoding=False, unlimited_dims=None
+        self, name, variable: Variable, check_encoding=False, unlimited_dims=None
     ):
         _ensure_no_forward_slash_in_name(name)
-
-        datatype = _get_datatype(
-            variable, self.format, raise_on_invalid_encoding=check_encoding
-        )
         attrs = variable.attrs.copy()
-
         fill_value = attrs.pop("_FillValue", None)
-
         encoding = _extract_nc4_variable_encoding(
             variable, raise_on_invalid=check_encoding, unlimited_dims=unlimited_dims
         )
-
+        if (
```

Comment:
Can we write this as a `Coder` in `coding/variables.py` please? That would be a lot cleaner.

Reply:
I've thought about that too, but did not find an immediate solution how to disentangle the in-file type creation.

Reply:
In an EnumCoder, we could turn the cf flag_* into a python Enum and move much of the checking code there.
Then in _netCDF4 module we can turn this python Enum into a netCDF4 enum when encoding is like `encoding["enum"] == "native"`.
But if it is `encoding["enum"] == "CF"` we could instead turn the enum into the cf flag_*.

What do you think ?

Reply:
Sounds reasonable, @bzah. How do you want to preserve/keep the enum name?

Reply:
I think we should simply use `"enum" in attrs` to decide to use the netCDF enum type. The `flag_*` stuff doesn't really need any special treatment.

I'm not a big fan of special-casing the `"enum"` attribute name, but I can't think of a better option at the moment.